src/transaction.rs

```diff
@@ -151,6 +151,7 @@ impl Config {
 pub struct Transaction<'conn> {
     conn: &'conn Connection,
     depth: u32,
+    savepoint_name: Option<&'conn str>,
```

Comment:
Sounds good to me. How's that?

I'll squash the commits down to one if/when this PR gets the thumbs up by the way.